contracts/staking/UniswapV2SwapHelper.sol

```diff
@@ -20,12 +16,13 @@ library UniswapV2SwapHelper {
 	 *@param _inTokenAmount amount of in token required to swap
 	 */
 	function maxSafeTokenAmount(
-		SimpleStaking staking,
+		IHasRouter _iHasRouter,
 		address _inToken,
 		address _outToken,
-		uint256 _inTokenAmount
+		uint256 _inTokenAmount,
+		uint256 _maxLiquidityPercentageSwap
```

Comment:
you can also keep this param
add it in the docs

Reply:
ADD DOCS everywhere

Reply:
when i say every where, i mean everywhere, not just where I commented.
for example in this same function ihasrouter is not in the function doc.
so go over all functions in uniswaphelper and in donation staking and make sure the docs matches